### PR TITLE
feat(admin-portal-petition): Admin portal module for "Undirskriftalistar"

### PR DESCRIPTION
**Pull request by alexdiljar**
Screenshot 2023-05-30 at 11 11 31
Screenshot 2023-05-30 at 11 11 40

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/portals/admin/src/lib/masterNavigation.ts [hunk 1]
```diff
@@ -34,6 +35,8 @@ export const TOP_NAVIGATION: PortalNavigationItem = {
     applicationSystemNavigation,
     // IDS Admin
     idsAdminNavigation,
+    // Undirskriftalistar
```
Comment (kksteini) [inline apps/portals/admin/src/lib/masterNavigation.ts:38]:
```suggestion
    // List of petitions
```

File: apps/services/endorsements/api/src/app/modules/endorsementList/endorsementList.service.ts [hunk 2]
```diff
@@ -44,7 +44,7 @@ export class EndorsementListService {
 
   async hasAdminScope(user: User): Promise<boolean> {
     for (const [_, value] of Object.entries(user.scope)) {
-      if (value == EndorsementsScope.admin) {
+      if (value == AdminPortalScope.petitionsAdmin) {
```
Comment (kksteini) [inline apps/services/endorsements/api/src/app/modules/endorsementList/endorsementList.service.ts:47]:
```suggestion
      if (value === AdminPortalScope.petitionsAdmin) {
```

File: libs/portals/admin/petition/src/components/ExportList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  const { formatMessage } = useLocale()
+  return (
+    <Box className={styles.buttonWrapper}>
+      <DropdownMenu
+        icon="download"
+        menuLabel={formatMessage(m.downloadPetitions)}
+        items={[
+          {
+            onClick: () => onGetCSV(),
+            title: 'Sem CSV',
```
Comment (alexdiljar) [inline libs/portals/admin/petition/src/components/ExportList/index.tsx:42]: Add to messages?

File: libs/portals/admin/petition/src/components/ListActions/UpdateList/UpdateList.action.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  const formData = await request.formData()
+
+  const openedDate = formData.get('openedDate') as string
+  const closedDate = formData.get('closedDate') as string
+
+  const description = formData.get('description') as string
+  const title = formData.get('title') as string
+
+  if (!params['listId']) {
+    throw new Error('something went wrong')
```
Comment (kksteini) [inline libs/portals/admin/petition/src/components/ListActions/UpdateList/UpdateList.action.ts:25]: Is this error shown to the user?
If not, can it be more specific for the sake of logs?
`Listid not provided or found in params`?

File: libs/portals/admin/petition/src/lib/messages.ts [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+  // Lock list
+  lockList: {
+    id: 'admin-portal.petition:lockList',
+    defaultMessage: 'Læsa lista',
+    description: '',
+  },
+  lockListMessage: {
+    id: 'admin-portal.petition:lockListMessage',
+    defaultMessage:
+      'Pellentesque tempor eros ut erat lacinia, eget tincidunt nibh viverra. Aenean scelerisque nulla purus, in gravida ipsum gravida vel.',
```
Comment (kksteini) [inline libs/portals/admin/petition/src/lib/messages.ts:133]: No text provided yet?

File: libs/portals/admin/petition/src/lib/messages.ts [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+  // Unlock list
+  unlockList: {
+    id: 'admin-portal.petition:unlockList',
+    defaultMessage: 'Aflæsa lista',
+    description: '',
+  },
+  unlockListMessage: {
+    id: 'admin-portal.petition:unlockListMessage',
+    defaultMessage:
+      'Pellentesque tempor eros ut erat lacinia, eget tincidunt nibh viverra. Aenean scelerisque nulla purus, in gravida ipsum gravida vel.',
```
Comment (kksteini) [inline libs/portals/admin/petition/src/lib/messages.ts:151]: Lorem?

File: libs/portals/admin/petition/src/screens/PetitionList/PetitionList.loader.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        input: {
+          listId: params.listId,
+        },
+      },
+    })
+    if (listError) {
+      throw listError
+    }
+    if (!listData) {
+      throw new Error('not found')
```
Comment (kksteini) [inline libs/portals/admin/petition/src/screens/PetitionList/PetitionList.loader.ts:33]: Both errors are `not found`.

If these are logged, is it better to be more specific, like?

```
throw new Error('Listid not provided in parameters')
...
...
throw new Error(`No list data found for ${params.listId}`)
```

**Review by alexdiljar — COMMENTED** [review]
One tiny thing, otherwise amazing

File: libs/portals/admin/petition/src/lib/messages.ts (hunk 6 above)
Comment (albinagu) [reply libs/portals/admin/petition/src/lib/messages.ts:151]: tbd

File: libs/portals/admin/petition/src/lib/messages.ts (hunk 5 above)
Comment (albinagu) [reply libs/portals/admin/petition/src/lib/messages.ts:133]: tbd

File: apps/portals/admin/src/lib/masterNavigation.ts (hunk 1 above)
Comment (albinagu) [reply apps/portals/admin/src/lib/masterNavigation.ts:38]: file is commented in icelandic

**Review by kksteini — COMMENTED** [review]
Nice!
Some semantic pedantry :)

**Review by kksteini — APPROVED** [review]

**Review by eirikurn — APPROVED** [review]
Approving core files

File: libs/portals/admin/petition/src/shared/queries/getAllEndorsementsLists.graphql [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    data {
+      id
+      title
+      description
+      closedDate
+      openedDate
+      adminLock
+    }
+  }
+}
```
Comment (eirikurn) [inline libs/portals/admin/petition/src/shared/queries/getAllEndorsementsLists.graphql:21]: Recommend colocating graphql queries and mutations where they're used, i.e. in the screen folder with the loader.

That way we can minimise sharing and easily strip out unneeded fields. Can even join together graphql fragments from subcomponents.

**Pull request merged**